lib/fastlane/plugin/ddg_apple_automation/actions/asana_report_failed_workflow_action.rb

```diff
@@ -0,0 +1,118 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "asana"
+require_relative "../helper/asana_helper"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "../helper/git_helper"
+require_relative "asana_add_comment_action"
+require_relative "asana_get_user_id_for_github_handle_action"
+
+module Fastlane
+  module Actions
+    class AsanaReportFailedWorkflowAction < Action
+      def self.run(params)
+        args = {
+          workflow_name: params[:workflow_name],
+          workflow_url: params[:workflow_url]
+        }
+
+        extra_collaborators = []
+
+        if params[:commit_sha]
+          args[:last_commit_url] = "https://github.com/#{Helper::GitHelper.repo_name}/commit/#{params[:commit_sha]}"
+          commit_author = Helper::GitHelper.commit_author(Helper::GitHelper.repo_name, params[:commit_sha], params[:github_token])
+          args[:last_commit_author_id] = Helper::AsanaHelper.get_asana_user_id_for_github_handle(commit_author)
+          extra_collaborators << args[:last_commit_author_id]
+        end
+
+        unless params[:is_scheduled_release]
+          args[:workflow_actor_id] = Helper::AsanaHelper.get_asana_user_id_for_github_handle(params[:github_handle])
+          extra_collaborators << args[:workflow_actor_id]
+        end
+
+        extra_collaborators.uniq!
+
+        assignee_id = Helper::AsanaHelper.extract_asana_task_assignee(params[:task_id], params[:asana_access_token])
+        if extra_collaborators.include?(assignee_id)
+          extra_collaborators.delete(assignee_id)
+        else
+          args[:assignee_id] = assignee_id
+        end
+
+        add_collaborators(extra_collaborators, params[:task_id], params[:asana_access_token])
```

Comment:
That's neat 🙌

Reply:
The sad thing about it is that it’s actually needed here, or otherwise people mentioned in the comment won’t get pinged in Asana 🤦 It turns out that when you mention someone in a comment (from UI) then they’re added to the task as collaborator. But this doesn’t happen when adding a comment using API and even though the comment properly mentions a team member, they don’t get a notification in the inbox unless they’re collaborators. That’s why we need to ensure they’re collaborating on the task before mentioning them in the comment...